sympy/core/numbers.py

```diff
@@ -920,7 +920,7 @@ class Float(Number):
     is_irrational = None
     is_number = True
 
-    is_real = True
+    is_extended_real = True
```

Comment:
@smichr  should this be is_real=True as well after #16727?

Reply:
It has not yet been decided whether Infinity should derive from Float or not (see https://github.com/sympy/sympy/pull/16727#issuecomment-487182859),

Reply:
@asmeurer says,

> In short, perhaps we can leave this as is for now, until a need for oo to be a Float becomes clear. It never happened with NaN, and maybe it will never happen with oo either.

And I suggested that extended real be `{-oo, Float, oo}` (implying that oo not be a Float). That 'inf' is represented by an mpf is not an argument that it should be a Float -- mpfs are "oo" aware just as SymPy is aware of it (as a Singleton).

I would vote to keep it as it is currently: `oo.is_Float -> False`.

It would make sense to tweak the Float instantiation to reject `nan`, `inf` and `-inf` in favor of instantiating those with `Number` instead since there is nothing to be gained by allowing them to instantiate from Float -- Float just returns the Singleton (which does not have the `_mpf_` attribute of Floats but can be coaxed to tell it as in `oo._as_mpf_val(1) -> (0, 0, -456, -2)`.

Reply:
@asmeurer also said:

>  I don't know if having oo be a true Float with an _mpf_ would make internal code cleaner, such as the code in evalf.py.

And I don't know whether that would make things better or not. But even still, we might consider a class `Real(Float)` or `ExtendedReal(Float)` which simply allows `_mpf_` to be requested directly instead of requiring special handling of `oo` with `_as_mpf_val`.

Reply:
I think that such classes might make the handling of real numbers even more confusing. There is already the `RealElement` class that is not derived from `Float` although both have attributes of `mpf` type.

Reply:
As to the question of `Float.is_real`, I think that it should be True. It seems that others concur.

Reply:
Thanks. I've re-added is_real=True so that Float.is_real is always True.